text/0739-ember-data-deprecate-non-strict-relationships.md

```diff
@@ -0,0 +1,195 @@
+---
+Stage: Accepted
+Start Date: 2021-04-23
+Release Date: Unreleased
+Release Versions:
+  ember-source: vX.Y.Z
+  ember-data: vX.Y.Z
+Relevant Team(s): ember-data
+RFC PR: https://github.com/emberjs/rfcs/pull/739
+---
+
+# EmberData | Deprecate Non Strict Relationships
+
+## Summary
+
+Deprecates various shorthands for defining a `belongsTo` or `hasMany` 
+relationship that create ambiguity, cause expensive runtime lookups, 
+or hinder static analysis.
+
+## Motivation
+
+Deprecating these shorthands allows us to provide better tooling, remove the bulky
+and expensive code necessary to determine the outcomes of these shorthands at 
+runtime, and in the process clarify and simplify relationship behaviors in the
+documentation. Further, it allows us to align our own internal usage of 
+"relationship meta" with the API put forward in [RFC#487 Custom Model Classes](https://github.com/emberjs/rfcs/blob/master/text/0487-custom-model-classes.md#exposing-schema-information).
+
+## Detailed design
+
+Three shorthands are proposed to be deprecated, with the deprecation targeting `5.0`
+and being `enabled` no-sooner than `4.1`. It may be `available` sooner.
+
+- Deprecates the relationship shorthand that does not provide the related type as the first argument.
+
+This deprecation is related to deprecating `non-strict` types as it requires us to
+normalize the property name into a type string in a manner which may not be correct.
+
+*Before*
+
+```./before.js
+import Model, { belongsTo, hasMany } from '@ember-data/model';
+
+export default class Comment extends Model {
+  @belongsTo()
+  author;
+
+  @hasMany({ async: true, inverse: null })
```

Comment:
I find it vaguely odd for the before to include both strict relationships (for this `hasMany`) and non-strict (for the `belongsTo` just above)